Source/BestProjectEver/Public/Character/BPE_Enemy.h

```diff
@@ -84,6 +84,14 @@ class BESTPROJECTEVER_API ABPE_Enemy : public ABPE_BaseCharacter, public IBPE_In
 	UPROPERTY(EditAnywhere, BlueprintReadOnly, Category="Patrol")
 	TObjectPtr<ABPE_PathFollowing> PatrolPathReference;
 
+	/** Handle for efficient management of destroy delay */
+	FTimerHandle TimerHandle_Destroy;
+	
+	/** Minimum time before actor is destroyed when is inactive (In the LobbyGameMode, inactive is when spawn pad stops
+	 * interacting with this actor or is death */
```

Comment:
I feel like you're coupling your lobby game mode to some of your actors too tightly. It's a bit hard to keep track of variables that only make sense in a specific game mode. For example, wherever you're handling gameplay logic, you have to be aware of what bIsInactive means when calling DropWeapon, even when it doesn't have anything to do with gameplay at all. Or if you want to use similar logic somewhere else, all your documentation is referencing the lobby game mode, so it wouldn't be clear that it is used in other cases.

We can discuss this in a call :)